File: 004/004.py
def is_palindrome(to_check):
    return str(to_check)[::-1] == str(to_check)


def largest_palindrome_product(last, first):
    largest = 0
    for a in range(last, first, -1):
        for b in range(last, first, -1):
            calculated = a * b
            if calculated > largest and is_palindrome(calculated):
                largest = calculated
    return largest

print(largest_palindrome_product(999, 100))

File: 004/004.php
<?php

function is_palindrome($orig) {
    $rev = 0;
    $n = $orig;
    while ($n > 0) {
        $rev = $rev * 10 + $n % 10;
        $n = intdiv($n, 10);
    }
    return $orig === $rev;
}

$largest = 0;
$last = 999;
$first = 100;
for ($a = $last; $a >= $first; $a--)
    for ($b = $last; $b >= $first; $b--) {
        $calculated = $a * $b;
        if ($calculated > $largest && is_palindrome($calculated))
            $largest = $calculated;
    }
echo "$largest\n";
